### ISSUE TITLE
h3 not recognised?

### ISSUE THREAD

**Issue by alvarotrigo** [11ty/eleventy#2026]
**Describe the bug**

When writting the .md file and using `### ` some headings are not recognised.

**To Reproduce**
Not quite sure. Sometimes they work, others they don't.

Having this:

```html
### Highlights Content
This is a demo

### Focuses on What Matters
This is a demo

### Faster loading times
This is a demo
```

Results in this:
![image](https://user-images.githubusercontent.com/1706326/137589693-b948d447-68d5-49d7-ae91-a0761601c291.png)
![image](https://user-images.githubusercontent.com/1706326/137589714-f588490d-63df-4117-8aa0-a44e15d179a3.png)

**Environment:**

- Mac OS
- "@11ty/eleventy": "^0.11.1"
-@11ty/eleventy-img": "^0.8.3",
    "@11ty/eleventy-plugin-rss": "^1.1.0",
    "@11ty/eleventy-plugin-syntaxhighlight": "^3.0.6",

**Comment by pdehaan** [comment]
Well, that’s rather interesting. Does it do that every time, or is it intermittent?
I’ve never seen that before, unless it’s related to the indentation issue where markdown is possibly converting parts of the page to “pre” tags. 

Do you have a repo somewhere that’s shows this behavior? I’d love to take a look.

**Comment by zachleat** [comment]
I think the first thing to try would be the 1.0 beta, https://www.11ty.dev/blog/eleventy-v1-beta/

**Comment by alvarotrigo** [comment]
I've fixed it with this extension for VS Code
https://marketplace.visualstudio.com/items?itemName=karlito40.fix-irregular-whitespace

Although perhaps this is something that 11ty could do on the build process too on MD files?
What do you guys think?

**Comment by pdehaan** [comment]
Nope, the initial snippet worked for me w/ 0.12.1 locally, so I can't see what character code is used in that particular H3 which is causing it to behave unexpectedly:

```html
<h3>Highlights Content</h3>
<p>This is a demo</p>

<h3>Focuses on What Matters</h3>
<p>This is a demo</p>

<h3>Faster loading times</h3>
<p>This is a demo</p>
```

Personally, I think trying to auto-fix irregular whitespace would probably outside the scope of Eleventy. It'd probably be pretty tricky to try and figure out when to try sanitizing whitespace and then have to add a flag to revert back to the current behavior if people were somehow expecting irregular whitespace in their project.
